src/bindings/python/tests/test_runtime/test_stateful.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import os
+import pytest
+
+from openvino import (
+    Core,
+    CompiledModel,
+    InferRequest,
+    Tensor,
+    compile_model,
+)
+
+from tests.utils.helpers import generate_model_with_memory
+from openvino.runtime.utils.types import get_dtype
+
+
+class Caller:
+    def __init__(self, base):
+        self.base = base
+
+    def query_state(self):
+        return self.base.query_state()
+
+    def reset_state(self):
+        return self.base.reset_state()
+
+    def call(self, *args, **kwargs):
+        if type(self.base) is CompiledModel:
+            return self.base(*args, **kwargs)
+        elif type(self.base) is InferRequest:
+            return self.base.infer(*args, **kwargs)
+        else:
+            raise RuntimeError("Unknown caller base!")
+
+
+@pytest.mark.parametrize("base_class",
+                         [CompiledModel,
+                          InferRequest])
```

Comment:
I remember that it is not good to use non-standard values in a list for parametrization from the point of parallelization of test execution. May be it is better to name it by string values and keep a map to class in the test body.

Reply:
I haven't heard about this tbh, could we merge it as-is and refactor later? BTW do you have any sources on that matter? I would like to learn more.